### PR TITLE
single_processing

### PR DESCRIPTION
**Pull request by fdcroce**

### CODE REVIEW

**Comment by fdcroce** [comment]
New function for applying perturbation using or not the multiprocessing. Valid only for cross section perturbations.
New function nuclide2latex for converting nuclide string to latex formatting.

File: sandy/zam.py [hunk 1]
```diff
@@ -856,3 +856,11 @@ def get_meta_letter(m, skip_ground=False):
...
         meta = ""
     return meta
+
+def nuclide2latex(nuclide, atomic_number=False, sep=""):
+    """
+    New function added to directly convert nuclides from string to latex
+    format useful for plotting.
+    """ 
+    out = zam2latex(nuclide2zam(nuclide, atomic_number = atomic_number, sep = sep))
+    return out
```
Comment (luca-fiorito-11) [inline sandy/zam.py]: add blank line at the end according to PEP8 (https://peps.python.org/pep-0008/)

File: sandy/zam.py [hunk 2]
```diff
@@ -856,3 +856,11 @@ def get_meta_letter(m, skip_ground=False):
     if skip_ground and m == 0:
         meta = ""
     return meta
+
+def nuclide2latex(nuclide, atomic_number=False, sep=""):
+    """
+    New function added to directly convert nuclides from string to latex
```
Comment (luca-fiorito-11) [inline sandy/zam.py]: It will not be "new" forever

File: sandy/zam.py [hunk 3]
```diff
@@ -856,3 +856,11 @@ def get_meta_letter(m, skip_ground=False):
...
+    format useful for plotting.
```
Comment (luca-fiorito-11) [inline sandy/zam.py]: remove "useful for plotting"

File: sandy/zam.py [hunk 4]
```diff
@@ -856,3 +856,11 @@ def get_meta_letter(m, skip_ground=False):
...
+    """ 
+    out = zam2latex(nuclide2zam(nuclide, atomic_number = atomic_number, sep = sep))
```
Comment (luca-fiorito-11) [inline sandy/zam.py]: no whitespaces in keyword arguments, according to PEP8

Comment (luca-fiorito-11) [reply sandy/zam.py]: I would write it as:
```
def nuclide2latex(nuclide, atomic_number=False, sep=""):
    out = zam2latex(nuclide2zam(*args, **kwargs))
    return out
```

This is not tested.
Also, Parameters and Returns missing in docstring

File: sandy/core/endf6.py [hunk 5]
```diff
@@ -2219,6 +2219,39 @@ def get_key(*lst):
         pool.join()
         return outs
 
+    def _mp_apply_perturbations_fdc(self, smps, processes,
```
Comment (luca-fiorito-11) [inline sandy/core/endf6.py]: naming is not consistent

Comment (luca-fiorito-11) [reply sandy/core/endf6.py]: remove this part from the PR. Let's merge the rest and then submit a new PR

File: sandy/zam.py (hunk 4 above)
Comment (luca-fiorito-11) [reply sandy/zam.py]: tests are also missing

**Review by luca-fiorito-11 — CHANGES_REQUESTED** [review]

**Comment by fdcroce** [comment]
i removed the function for applying perturbation since not compliant and I modified to nuclide2latex to implement the recommendations.

**Review by luca-fiorito-11 — APPROVED** [review]

**Comment by fdcroce** [comment]
Fix bugs in the examples.

**Comment by luca-fiorito-11** [comment]
@fdcroce , can you resubmit a PR only including `nuclide2latex`? Thanks

**Pull request closed without merging**